src/main/java/org/oscii/neural/Word2VecManager.java

```diff
@@ -138,6 +138,12 @@ public boolean rankConcordances(String lang, String context, List<SentenceExampl
         double sim = VectorMath.cosineSimilarity(tokensMean, contextMean);
         if (Double.isNaN(sim)) {
           sim = -2.0; // Give it a low score.
+        } else if (ex.memoryId == memoryId) {
+          // personal TM match in same project: place on top by adding +2.0 (note: similarity is not cosine sim any more)
+          sim += 2.0;
+        } else if (ex.memoryId > 0) {
```

Comment:
Cool. Merge at will.

On Tue, Jul 19, 2016 at 9:37 AM, Saša Hasan <notifications@github.com>
wrote:

> In src/main/java/org/oscii/neural/Word2VecManager.java
> <https://github.com/oscii-lab/lex/pull/20#discussion_r71375471>:
>
> > @@ -138,6 +138,12 @@ public boolean rankConcordances(String lang, String context, List<SentenceExampl
> >          double sim = VectorMath.cosineSimilarity(tokensMean, contextMean);
> >          if (Double.isNaN(sim)) {
> >            sim = -2.0; // Give it a low score.
> > +        } else if (ex.memoryId == memoryId) {
> > +          // personal TM match in same project: place on top by adding +2.0 (note: similarity is not cosine sim any more)
> > +          sim += 2.0;
> > +        } else if (ex.memoryId > 0) {
>
> somewhere in lex, memory got initialized to 0 (I think in the Request
> class), so I didn't know if this was maybe some special id; the mysql
> tables always start their index at 1 with default auto_increment, so checks
> against > 0 are ok I think; I'll leave as-is.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/oscii-lab/lex/pull/20/files/a674a2e9af707be657d57afb37df7f6730ae3385#r71375471>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AATikTynkabNl2lu6u8_PIkUkVzEMf_pks5qXP0zgaJpZM4JPjsz>
> .
>